packages/eas-cli/src/commands/abstract/authorizedCommand.ts

```diff
@@ -0,0 +1,18 @@
+import { Command } from '@oclif/command';
+
+import { flushAsync, initAsync } from '../../analytics';
+import { ensureLoggedInAsync } from '../../user/actions';
+
+export default abstract class AuthorizedCommand extends Command {
```

Comment:
AuthenticatedCommand if you want to name it this way, but it's still not really correct, if you are also handling analytics in the class

Maybe use some general  name like 
```suggestion
import { Command as OclifCommand } from '@oclif/command';

import { flushAsync, initAsync } from '../../analytics';
import { ensureLoggedInAsync } from '../../user/actions';

export default abstract class Command extends OclifCommand {
```

and unless that `abstract` directory is sth defined by oclif change it to sth like `wrapper`, 'utils`, `command` or maybe even remove that directory

Reply:
I'm not opposed to a name change but I feel like something less descriptive, like `Command`, leaves us worse off. Not every command requires authentication, so what do we call those commands that don't use this class? Also `Command`, but from oclif? seems confusing.

Reply:
> I'm not opposed to a name change but I feel like something less descriptive, like Command, leaves us worse off. Not every command requires authentication, so what do we call those commands that don't extend this class? Probably also Command? seems confusing.

I think it's more confusing if name is specific, but it has behavior unrelated to that name, `Command` was just an example we can just as well use sth that is still generic but different from oclif version. `CustomCommand`, `BaseCommand`, `EasCommand`,`DecoratedCommand`.

Currently, almost all commands require login and few that do not are more or of an exception here. I'm not sure if there is a need to future proof the code, eas-cli is a tool for interacting with eas services almost everything that does not require communication with www should be in expo-cli and I think in the future that requirement will only apply to more commands

>  think I'd prefer a more descriptively named folder to keep this class inside of if that works for you. Maybe something like abstractCommands or commandsBase.

I have a problem with word `abstract` 
- if you refereed to  `abstract class` then it does not tell anything about the the behavior, it just specifies feature of the language that was used there, which does not tell anything about the content. 
- if you refereed to this as an abstraction, then i don't think it's correct

`utils` or `wrapper` at least suggest that code is used in implementation of other commands, if you want something, more descriptive then maybe `./src/commandUtils/BaseCommand.ts`?

Reply:
> I think it's more confusing if name is specific, but it has behavior unrelated to that name, Command was just an example we can just as well use sth that is still generic but different from oclif version. CustomCommand, BaseCommand, EasCommand,DecoratedCommand.

it's a fair point. I think maybe the takeaway is to write a class that can be subclassed by all our commands, rather than just a subset.  `EasCommand` would be a pretty good descriptor for that, imo.   

> if you refereed to abstract class then it does not tell anything about the the behavior, it just specifies feature of the language that was used there, which does not tell anything about the content.

I actually think this is ideal, because it's a folder name. the file name should describe the behavior, the folder should describe _what_ is in there. So something like `./src/abstractCommands/EasCommand.ts` i think is quite descriptive, but i think`./src/commandUtils/BaseCommand.ts` is pretty good as well. I don't think it's too big a deal so I'll just go ahead and make that change.